benchmarks/apps/gromacs/gromacs.py

```diff
@@ -0,0 +1,98 @@
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+from benchmarks.modules.utils import SpackTest
+
+
+input_data_file = 'gromacs_1400k_atoms.tpr'
+
+class GROMACSBenchmark(SpackTest):
+    """Base class for a GROMACS benchmark using the spack build system"""
+
+    valid_systems = ['*']
+    valid_prog_environs = ['default']
+
+    # Variables consistent in all tests
+    time_limit = '45m'
+
+    expected_output_file = 'md.log'
+    keep_files = [expected_output_file]
+    readonly_files = [input_data_file]
+    sourcesdir = os.path.dirname(__file__)
+    executable = 'gmx_mpi'
+    
+    reference = {
+        'tursa:gpu-a100-40': {
+            'Rate': (6.8, -0.1, None, 'ns/day'),
+            'Energy': (-12070100.0, -1.0, 1.0, 'kJ/mol')
+        },
+        'kathleen:compute-node': {
+            'Rate': (1, -0.1, None, 'ns/day'),
+            'Energy': (-12070100.0, -1.0, 1.0, 'kJ/mol')
+        },
+        '*': {
+            'Rate': (1, None, None, 'ns/day'),
+            'Energy': (1, None, None, 'kJ/mol')
+        }
+    }
+        
+    @run_after('setup')
+    def setup_test_variables(self):
+        """Set the variables required after setup, specific to each test"""
+        self.num_tasks_per_node = int(self.current_partition.processor.num_cpus / self.current_partition.processor.num_cpus_per_core)
+        self.num_tasks = self.num_tasks_per_node * self.num_nodes_param
+        self.num_nodes = self.num_nodes_param
+        self.num_cpus_per_task = 1
+        
+        self.env_vars['OMP_NUM_THREADS'] = f'{self.num_cpus_per_task}'
+        self.env_vars['OMP_PLACES'] = 'cores'
+
+    @run_before('sanity')
+    def set_test_sanity_patterns(self):
+        """Set the required string in the output for a sanity check"""
+        self.sanity_patterns = sn.assert_found(
+            'Finished mdrun', self.expected_output_file
+        )
+
+    @run_before('performance')
+    def set_test_perf_patterns(self):
+        """Set the regex performance pattern to locate"""
+        self.perf_patterns = {
+            'Rate': sn.extractsingle(r'Performance:\s+(?P<rate>\S+)(\s+\S+){1}',
+                                     self.expected_output_file, 'rate', float),
+            'Energy': sn.extractsingle(r'\s+Potential\s+Kinetic En\.\s+Total Energy\s+Conserved En\.\s+Temperature\n'
+                                       r'(\s+\S+){2}\s+(?P<energy>\S+)(\s+\S+){2}\n'
+                                       r'\s+Pressure \(bar\)\s+Constr\. rmsd',
+                                       self.expected_output_file, 'energy', float),
+        }
+
+@rfm.simple_test
+class StrongScalingCPU(GROMACSBenchmark):
+    spack_spec = 'gromacs@2024 +mpi'
+    tags = {"cpu"}
+
+    executable_opts = ['mdrun', '-noconfout', '-dlb', 'yes', '-s', input_data_file]
+    num_nodes_param = parameter([1, 2, 3, 4])
+
+
+@rfm.simple_test
+class StrongScalingSpackGPU(GROMACSBenchmark):
+    spack_spec = 'gromacs@2024 +mpi+cuda'
+    tags = {"gpu"}
```

Comment:
Tags are good to have, but shouldn't you use the `valid_systems` variable so that the GPU benchmark won't run on a system without a GPU and vice versa? For example you can have a look at babelstream https://github.com/ukri-excalibur/excalibur-tests/blob/f095234cd131161a5324ed374fcc0ad785807dbb/benchmarks/apps/babelstream/babelstream.py#L60-L74